src/main/java/org/rsa/command/CommandObject.java

```diff
@@ -19,32 +19,46 @@ public abstract class CommandObject extends ListenerAdapter {
     private final String name;
     private final String description;
     private final List<OptionData> optionDataList;
+    private final List<SubcommandObject> subcommandList;
     private boolean isAutocomplete = false;
+    private boolean isGuildOnly = false;
 
     public CommandObject(String name, String description) {
-        this(name, description, Collections.emptyList());
+        this(name, description, Collections.emptyList(), Collections.emptyList());
     }
 
-    public CommandObject(String name, String description, List<OptionData> optionDataList) {
+    public CommandObject(String name, String description, List<OptionData> optionDataList)
+    {
+        this(name, description, optionDataList, Collections.emptyList());
+    }
+
+    public CommandObject(String name, String description, List<OptionData> optionDataList, List<SubcommandObject> subcommandList) {
         this.name = name;
         this.description = description;
         this.optionDataList = new ArrayList<>(optionDataList);
+        this.subcommandList = new ArrayList<>(subcommandList);
     }
 
     public void addOptionData(OptionData optionData) {
         optionDataList.add(optionData);
     }
 
+    public void addSubcommand(SubcommandObject subcommandObject) { subcommandList.add(subcommandObject); }
+
     public void setIsAutocomplete() {
         isAutocomplete = true;
     }
 
+    public void setIsGuildOnly() {
+        isGuildOnly = true;
+    }
+
     public List<OptionData> getOptionDataList() {
         return new ArrayList<>(optionDataList);
     }
 
     public SlashCommandData slashCommandImplementation() {
-        return Commands.slash(name, description).addOptions(optionDataList);
+        return Commands.slash(name, description).addOptions(optionDataList).addSubcommands(subcommandList).setGuildOnly(isGuildOnly);
```

Comment:
I don't think we should `setGuildOnly` here. There may be a case where we want a command to function in DMs.